server/controllers/paymentDetails/paymentDetails.controller.js

```diff
@@ -0,0 +1,60 @@
+import User from '../../models/user';
+import CreditCard from '../../models/creditCard';
+
+const getMyPaymentDetails = function(req, res) {
+  User.
+  findById(req.userId)
+    .select('creditCard')
+    .populate('creditCard')
+    .exec((err, user) => {
+      if (err) {
+        res.status(500).send(err);
+      } else if (!user) {
+        res.status(404).send();
+      } else {
+        res.status(200).send(user);
+      }
+    });
+};
+
+const changePaymentDetails = function(req, res) {
```

Comment:
can we have 2 separate methods here, one for creation and one for updating
POST for creation
 -- find user by id
 -- create CreditCard
 -- save CreditCard
 -- set user.CreditCard
 -- save user
and PUT/:id for updating (the getMyPaymentDetails returns the whole credit card, so we have an id)
 -- CreditCard.findById
 -- update fields
 -- save changes

that way we don't have to do an inline deletion and its only 1 save